extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/ContainerImageConfig.java

```diff
@@ -102,6 +106,9 @@
         if (group.isPresent()) {
             String originalGroup = group.get();
             if (originalGroup.equals(System.getProperty("user.name"))) {
+                if (originalGroup.isEmpty()) {
```

Comment:
Hmmm. So the converter makes the string always present?

Reply:
Seems to be the case...

Reply:
If it is, that's definitely not what I expected. I would have expected the Optional to stay empty if no string is defined. Let's wait for @radcortez 's opinion on this :).

Reply:
Yeah, I agree

Reply:
The issue here is that the Converter must return `null` if there is no value. From the Config rules, an empty String `""` is automatically converted to `null`, but in the case of this `TrimmedStringConverter` the conversion is the empty String again, which is then passed to the Optional, so you get a non empty Optional.

The `TrimmedStringConverter` must return a `null` if the resulting String is an empty String. Now we may need to recheck other usages of this converter to make sure we don't break it.